Engines/Wine/Engine/Implementation/script.js

```diff
@@ -324,6 +324,12 @@ var engineImplementation = {
             return this.run("start", batArgs, workingDir, captureOutput, wait, userData);
         }
 
+        if (userData["trustLevel"] == "0x20000" && distribution == "staging") {
+            var runasArgs = org.apache.commons.lang.ArrayUtils.addAll(["/trustlevel:0x20000", executable], args);
```

Comment:
This line is really ugly. Please split it in two:

```javascript
const ArrayUtils = Java.type("org.apache.commons.lang.ArrayUtils");
const runAsArgs = ArrayUtils.addAll(["/trustlevel:0x20000", executable], args);
```

@plata @qparis @ImperatorS79 ideally we do this change for every usage of Java classes and move the `const JavaClass = Java.type(...);` parts to the beginning of the files directly beneath the includes.

Reply:
IMHO, Ideally, we should avoid calling directly java objects and use beans whenever it is possible.
This would make the scripts more agnostic of the main application

Reply:
I have just copied the few lines above ^^.

Reply:
Isn't directly calling Java objects basically the same as using (Java) beans? Both are Java objects underneath with the only difference that beans do not require `Java.type(...)`.

If we want to do this in a clean way we will need to remove any kind of Java objects/classes from the script repository and access JavaScript libraries instead. If we do this the only interaction between the Java side and the Script side would be through method invocations **from** the Java side.

Reply:
Two advantages of using beans: 
- It could be implemented in any language :) Basically it would just represent a method that gets a property from an application context map. It makes our scripts more portable and testable in a future. 
- Later, we can take advantage of it to add security feature. (I.e. block all access to classes outside phoenicis)

Reply:
> * It could be implemented in any language :) Basically it would just represent a method that gets a property from an application context map. It makes our scripts more portable and testable in a future.

Aren't the scripts equally portable if it only depends on source code and libraries written in JavaScript?

> Later, we can take advantage of it to add security feature. (I.e. block all access to classes outside phoenicis)

If we don't use any kind of Java objects from inside the scripts, we can simply forbid access to all kinds of Java objects from inside the scripts. I believe this would lead to the best security, right?

Reply:
Not totally. I would prefer Native.type instead of Java.type. In addition, and more important than the naming in my point of view: When you allow guest classes, you may need to reimplement the universe to port your script. If we have a clear and managed set of Beans interface you have to implement, it is a lot easier. 

> If we don't use any kind of Java objects from inside the scripts, we can simply forbid access to all kinds of Java objects from inside the scripts. I believe this would lead to the best security, right?
Forbid acess to all kinds of Java objects, except beans that handle correctly the security (like filesystem access, etc...)

Reply:
But I think it is something we could discuss about in another issue

Reply:
Yes. Still for now I would prefer to move all done Java class imports to the beginning of the script files. This also makes it easier for us to find the Java object usages later on, because they are all in one place.

Reply:
Sounds fair

Reply:
@ImperatorS79 can you please try 

```javascript
const runAsArgs = ["/trustlevel:0x20000", executable].concat(args)
```

alternatively? This will avoid the Java dependency totally.